tests/v2/validation/rbac/globalrolesv2/globalroles_v2_webhook_test.go

```diff
@@ -0,0 +1,315 @@
+		if newLabels == nil {
+			newLabels = make(map[string]string)
+		}
+		delete(newLabels, crtbOwnerLabel)
+
+		crtbWithUpdatedLabels := &v3.ClusterRoleTemplateBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        crtb.Name,
+				Namespace:   crtb.ObjectMeta.Namespace,
+				Labels:      newLabels,
+				Annotations: crtb.Annotations,
+			},
+			ClusterName:       crtb.ClusterName,
+			UserName:          crtb.UserName,
+			RoleTemplateName:  crtb.RoleTemplateName,
+			UserPrincipalName: crtb.UserPrincipalName,
+		}
+
+		_, err = rbac.UpdateClusterRoleTemplateBindings(grw.client, crtb.Namespace, crtbWithUpdatedLabels)
+		require.Error(grw.T(), err)
+
+		expectedErrMessage := "admission webhook \"rancher.cattle.io.clusterroletemplatebindings.management.cattle.io\" denied the request: clusterroletemplatebinding.labels: Forbidden: label authz.management.cattle.io/grb-owner is immutable after creation"
+		assert.Contains(grw.T(), err.Error(), expectedErrMessage)
+
+	}
+
+}
+
+func (grw *GlobalRolesV2WebhookTestSuite) TestLockedRoleTemplateInInheritedClusterRole() {
+
+	log.Info("Verify adding a locked custom cluster role template is rejected by webhook.")
+	inheritedRoleTemplateName := namegen.AppendRandomString("customrole")
+
+	lockedRoleTemplate, err := rbac.CreateRoleTemplate(grw.client, inheritedRoleTemplateName, clusterContext, []string{roleOwner},
+		nil, nil, nil, false, true)
+	require.NoError(grw.T(), err)
+
+	_, err = createGlobalRole(grw.client, []string{lockedRoleTemplate.Name})
+	require.Error(grw.T(), err)
+
+	start := "admission webhook"
+	end := "unable to use locked roleTemplate"
+
+	getError := getErrorMessage(err.Error(), start, end)
+
+	expectedErrMessage := "admission webhook \"rancher.cattle.io.globalroles.management.cattle.io\" denied the request: globalrole.inheritedClusterRoles: Invalid value: \"" + lockedRoleTemplate.Name + "\": unable to use locked roleTemplate"
+	assert.Contains(grw.T(), expectedErrMessage, getError)
+
+}
+
+func (grw *GlobalRolesV2WebhookTestSuite) TestAddGlobalRoleWithCustomTemplateAndLockRoleTemplate() {
+
+	log.Info("Verify adding a custom cluster role template and then locking the template is not rejected by webhook.")
+	roleTemplateName := namegen.AppendRandomString("customrole")
+
+	customRoleTemplate, err := rbac.CreateRoleTemplate(grw.client, roleTemplateName, clusterContext, []string{roleOwner}, nil, nil, nil, false, false)
+	require.NoError(grw.T(), err)
+
+	_, err = createGlobalRole(grw.client, []string{customRoleTemplate.Name})
+	require.NoError(grw.T(), err)
+
+	customRoleTemplate.Locked = true
+	lockCustomRoleTemplate, err := rbac.UpdateRoleTemplate(grw.client, customRoleTemplate)
+	require.NoError(grw.T(), err)
+	assert.Equal(grw.T(), lockCustomRoleTemplate.Locked, true)
+
+}
+
+func (grw *GlobalRolesV2WebhookTestSuite) TestDeleteCustomRoleTemplateInInheritedClusterRole() {
+	log.Info("Verify deleting a custom cluster thats inherited by global role is rejected.")
```

Comment:
Did you intend to mention "deleting a custom role template" and not "custom cluster"?